audit/logger.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2017 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/control-center/serviced/datastore"
+	"github.com/control-center/serviced/logging"
+	"github.com/zenoss/logri"
+)
+
+var plog = logging.PackageLogger()
+
+// Logger is the interface for audit logging.  Any implementations for
+// audit logging should implement this interface.
+type Logger interface {
+
+	// Set the action that we are auditing.
+	Action(action string) Logger
+
+	// Set the message that we are writing to the audit log.
+	Message(ctx datastore.Context, message string) Logger
+
+	// Set the type of entity being modified.
+	Type(theType string) Logger
+
+	// Set the id of the entity being modified.
+	ID(id string) Logger
+
+	// Set the type of entity being modified.
+	Entity(entity datastore.Entity) Logger
+
+	// Add additional fields to the entry.
+	WithField(name string, value string) Logger
+
+	// Log that the action succeeded.
+	Succeeded()
+
+	// Log that the action failed.
+	Failed()
+
+	// Log whether the action succeeded or failed based on the value passed in.
+	SucceededIf(value bool)
+
+	// Log whether the action succeeded or failed based on the error passed in.
+	Error(err error) error
+}
+
+// NewLogger returns a default implmentation of the audit logger.  The "user" will default
+// to "system" unless otherwise specified in the context.  This wraps a logri Logger,
+// and will write to the location specified in the logger config.
+func NewLogger() Logger {
+	entry := logri.GetLogger("audit").WithField("user", "system")
```

Comment:
I don't think we need to the user field here. I think we should let the `Message()` method always get the user from the context, and let the context create the default user. Basically, I don't want the default user to be defined in more than one place - the context object should be the system of record for that.